src/vmprof_common.h

```diff
@@ -108,9 +114,52 @@ static int opened_profile(char *interp_name, int memory)
  
 PyThreadState* get_current_thread_state(void)
 {
+    PyThreadState *tstate;
 #if PY_MAJOR_VERSION >= 3
-    return (PyThreadState*)_Py_atomic_load_relaxed(&_PyThreadState_Current);
+    tstate = (PyThreadState*)_Py_atomic_load_relaxed(&_PyThreadState_Current);
 #else
-    return _PyThreadState_Current;
+    tstate = _PyThreadState_Current;
 #endif
+
+    if (tstate == NULL) {
```

Comment:
It isn't unrelated. The problem is that `tstate` is always `NULL` when waiting for a syscall, so if we didn't look for the thread in the interpreter state, we'd never get any profiling samples for the program while it's waiting for a syscall. It wouldn't make sense to use `ITIMER_REAL` without this code.